From 71a03076625286349103d34d8d93bc5fa3e68d92 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Tue, 5 Jun 2018 00:53:17 +0800
Subject: [PATCH 3940/5242] MGS-3940-1 gpu: imx: dpu-blit: fix first frame
 handler

commit  809d058aae60838a2ae943f4884a535ea18dc613 from
https://source.codeaurora.org/external/imx/linux-imx.git

chrome browser hang is reproduced with mouse connected,
the first frame handler trigger the problem in below scenario:
tile (dprc-enable) --> linear (dprc->disable) --> tile (handle first frame wrongly).

need_handle_start is set following dprc_enable, need reset it with dprc_disable.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu-blit/dpu-blit.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu-blit/dpu-blit.c b/drivers/gpu/imx/dpu-blit/dpu-blit.c
index b996502..8c2c5f8 100644
--- a/drivers/gpu/imx/dpu-blit/dpu-blit.c
+++ b/drivers/gpu/imx/dpu-blit/dpu-blit.c
@@ -127,8 +127,10 @@ void dpu_be_configure_prefetch(struct dpu_bliteng *dpu_be,
 	}
 
 	if (baddr == 0x0) {
-		if (!start)
+		if (!start) {
 			dprc_disable(dprc);
+			need_handle_start = false;
+		}
 		start = true;
 		return;
 	}
-- 
1.7.9.5

